Source/WebCore/rendering/svg/RenderSVGResourceClipper.cpp

```diff
@@ -191,6 +191,24 @@ void RenderSVGResourceClipper::applyMaskClipping(PaintInfo& paintInfo, const Ren
     frameView.setPaintBehavior(oldBehavior);
 }
 
+FloatRect RenderSVGResourceClipper::calculateClipContentRepaintRect(RepaintRectCalculation repaintRectCalculation)
+{
+    FloatRect clipContentRepaintRect;
+    // This is a rough heuristic to appraise the clip size and doesn't consider clip on clip.
+    for (auto* childNode = clipPathElement().firstChild(); childNode; childNode = childNode->nextSibling()) {
+        auto* renderer = childNode->renderer();
+        if (!childNode->isSVGElement() || !renderer)
+            continue;
+        if (!renderer->isRenderOrLegacyRenderSVGShape() && !renderer->isRenderSVGText() && !childNode->hasTagName(SVGNames::useTag))
+            continue;
+        auto& style = renderer->style();
+        if (style.display() == DisplayType::None || style.visibility() != Visibility::Visible)
+            continue;
+        clipContentRepaintRect.unite(renderer->repaintRectInLocalCoordinates(repaintRectCalculation));
```

Comment:
Need to check, if the repaint rect accessor usage  is correct.

Reply:
Seems fine - this includes the transformations of the children, which is what we need here.